clash-prelude/src/Clash/Explicit/BlockRam/Blob.hs

```diff
@@ -88,7 +88,9 @@ import Clash.XException
 blockRamBlob
   :: forall dom addr m n
    . ( KnownDomain dom
+     , KnownNat m
```

Comment:
This `KnownNat m` should not be necessary. If for some reason type checking does want this, it should be done by pattern matching on the `MemBlob` constructor instead. But it seems odd that the introduction of `NFDataX` would necessitate this addition. I can look at it tomorrow.

Reply:
Right, I get it! Yes, the change does add a `KnownNat`, because the `NFDataX` instance for `BitVector` adds that constraint. It's late :-).

Reply:
Done, I had just blindly followed the types!